capstone_project/src/main/java/com/google/sps/data/PostService.java

```diff
@@ -107,6 +120,11 @@ public long upvotePost(HttpServletRequest request) {
 
     List<Entity> posts = datastore.prepare(query).asList(FetchOptions.Builder.withDefaults());
 
+    String sortType = request.getParameter("sort-type");
```

Comment:
could this be null?

Reply:
Technically it can, but the http request will always have a sort-type, unless a user modifies the html via the browser console.